src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala

```diff
@@ -84,9 +84,17 @@ object DelimitedDataParser {
   def apply(path: FilePath, delimiter: Char): DelimitedDataParser =
     new DelimitedDataParser(Io.toSource(path).getLines(), delimiter=delimiter)
 
+  /** Constructs a DelimitedDataParser for a path. */
+  def apply(path: FilePath, delimiter: Char, header: Seq[String]): DelimitedDataParser =
+    new DelimitedDataParser(Io.toSource(path).getLines(), delimiter=delimiter, header=header)
```

Comment:
It's probably good to re-route the apply on 84-85 to use this now one with `header=Seq.empty`. Ditto below.

Reply:
Scala will not allow me to have multiple overloaded alternatives of the same method with default arguments defined:

https://stackoverflow.com/a/10138519/3727678

https://gist.github.com/lossyrob/63f0d40506ff8b4c3b22

Reply:
I am not asking to have default arguments in the function signature.  I am asking to have the `apply` method _without_ the `header` parameter call the `apply` method _with_ the `header` parameter, so if we change the call to the latter, the former does not need updating. For example:

```scala
  /** Constructs a DelimitedDataParser for a path. */
  def apply(path: FilePath, delimiter: Char, header: Seq[String]): DelimitedDataParser =
    new DelimitedDataParser(Io.toSource(path).getLines(), delimiter=delimiter, header=header)

  /** Constructs a DelimitedDataParser for a path. */
  def apply(lines: TraversableOnce[String], delimiter: Char): DelimitedDataParser =
    this.apply(lines, delimiter=delimiter, header=Seq.empty)
```

Reply:
![oh-yea-duh](https://user-images.githubusercontent.com/4634699/57328150-80021880-70c5-11e9-9a1b-ec323df40a70.gif)

Reply:
![64CC31C0-60DF-4ABE-BE37-622B3298625F](https://user-images.githubusercontent.com/846274/57329392-880f8780-70c8-11e9-9a48-6f9d946c66e2.gif)

Reply:
@clintval I think L89 can be:
```suggestion
    apply(Io.toSource(path).getLines(), delimiter=delimiter, header=header)
```